control-plane/connect-inject/endpoints_controller.go

```diff
@@ -661,7 +660,7 @@ func getHealthCheckStatusReason(healthCheckStatus, podName, podNamespace string)
 // The argument endpointsAddressesMap decides whether to deregister *all* service instances or selectively deregister
 // them only if they are not in endpointsAddressesMap. If the map is nil, it will deregister all instances. If the map
 // has addresses, it will only deregister instances not in the map.
-func (r *EndpointsController) deregisterServiceOnAllAgents(ctx context.Context, k8sSvcName, k8sSvcNamespace string, endpointsAddressesMap map[string]bool, endpointPods mapset.Set) error {
+func (r *EndpointsController) deregisterServiceOnAllAgents(ctx context.Context, k8sSvcName, k8sSvcNamespace string, endpointsAddressesMap map[string]bool, endpointPods mapset.Set) (ctrl.Result, error) {
```

Comment:
Looks like this method always returns only `ctrl.Result{}`. IMO we should undo this refactor as it makes the method signature confusing.

Reply:
Makes sense. I like how clean this makes calling the method, but I get your concern.